src/main/java/uk/gov/hmcts/reform/prl/services/managedocuments/ManageDocumentEmailService.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.hmcts.reform.prl.services.managedocuments;
+
+import com.sendgrid.Method;
+import com.sendgrid.Request;
+import com.sendgrid.Response;
+import com.sendgrid.SendGrid;
+import com.sendgrid.helpers.mail.Mail;
+import com.sendgrid.helpers.mail.objects.Email;
+import com.sendgrid.helpers.mail.objects.Personalization;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.MapUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.prl.config.SendgridEmailTemplatesConfig;
+import uk.gov.hmcts.reform.prl.enums.LanguagePreference;
+import uk.gov.hmcts.reform.prl.models.email.SendgridEmailConfig;
+import uk.gov.hmcts.reform.prl.models.email.SendgridEmailTemplateNames;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.prl.services.SendgridService.MAIL_SEND;
+import static uk.gov.hmcts.reform.prl.services.SendgridService.NOTIFICATION_TO_PARTY_SENT_SUCCESSFULLY;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ManageDocumentEmailService {
+
+    private final SendgridEmailTemplatesConfig sendgridEmailTemplatesConfig;
+
+    private final SendGrid sendGrid;
+    @Value("${send-grid.rpa.email.from}")
+    private String fromEmail;
+
+
+    public void sendEmailUsingTemplateWithAttachments(SendgridEmailTemplateNames sendgridEmailTemplateNames,
```

Comment:
Can we have an Async event handler for this, given this is only sending out emails